source/mqtt_client_connection.c

```diff
@@ -40,12 +40,12 @@ struct mqtt_connection_binding {
 };
 
 static void s_mqtt_client_connection_release_threadsafe_function(struct mqtt_connection_binding *binding) {
-    AWS_NAPI_ENSURE(
-        binding->env, aws_napi_release_threadsafe_function(binding->on_connection_interrupted, napi_tsfn_abort));
-    AWS_NAPI_ENSURE(
-        binding->env, aws_napi_release_threadsafe_function(binding->on_connection_resumed, napi_tsfn_abort));
-    AWS_NAPI_ENSURE(binding->env, aws_napi_release_threadsafe_function(binding->on_any_publish, napi_tsfn_abort));
-    AWS_NAPI_ENSURE(binding->env, aws_napi_release_threadsafe_function(binding->transform_websocket, napi_tsfn_abort));
+    AWS_NAPI_CALL(
+                  binding->env, aws_napi_release_threadsafe_function(binding->on_connection_interrupted, napi_tsfn_abort), ;);
+    AWS_NAPI_CALL(
+                  binding->env, aws_napi_release_threadsafe_function(binding->on_connection_resumed, napi_tsfn_abort), ;);
+    AWS_NAPI_CALL(binding->env, aws_napi_release_threadsafe_function(binding->on_any_publish, napi_tsfn_abort), ;);
+    AWS_NAPI_CALL(binding->env, aws_napi_release_threadsafe_function(binding->transform_websocket, napi_tsfn_abort), ;);
 }
 
 static void s_mqtt_client_connection_finalize(napi_env env, void *finalize_data, void *finalize_hint) {
```

Comment:
(comment not related to this line)
`builder.json` has this line: `CDCMAKE_PREFIX_PATH={source_dir}/{libcrypto_path}`

Can this be removed? Is it  magic for finding the old openssl libcrypto?

Reply:
Umm...? Maybe? I'd rather not discover where it might break with this change if it has been there for a while

Reply:
my argument is: I'm pretty sure it's there to support finding openssl's libcrypto
and this PR is trying to remove dependence on openssl's libcrypto

Reply:
Yeah, remove it. That's for builder installed libcrypto.

Reply:
Looking at the build logs, it seems that aws-lc is being compiled twice:
```
2021-03-18T18:06:40.9977479Z [ 80%] Linking C static library libcrypto.a
...
2021-03-18T18:08:37.1974207Z [ 62%] Linking C static library libcrypto.a
```

also, there's an error the second time it tries to build... a bit worrisome that this doesn't kill the build:
```
2021-03-18T18:07:29.7519911Z -- Pre aws-lc CMAKE_INSTALL_PREFIX=/usr/local
2021-03-18T18:07:29.7556672Z CMake Error: The source "/root/aws-crt-nodejs/crt/aws-lc/CMakeLists.txt" does not match the source "/root/aws-crt-nodejs/deps/aws-lc/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.
2021-03-18T18:07:29.7565801Z CMake Error at crt/aws-c-cal/cmake/modules/aws-lc.cmake:22 (message):
2021-03-18T18:07:29.7566707Z   Failed to configure aws-lc
```

I think the builder is trying to do magic to build aws-lc for you, but it doesn't need to.
Probably the magic is from these lines in builder.json, which should be removed:
```
            "imports": [
                "s2n"
            ],
```